### ISSUE TITLE
Question on dataset GraspNet

### ISSUE THREAD

**Issue by crepuscularlight** [donydchen/mvsplat#67]
Thank you first for your great work. 
I try to adapt this model to dataset graspnet. And I follow the steps in pixelsplat to write the data loader. It works normally on pixelsplat but the training on mvsplat does not converge. Could you give me some advice how to debug this?

**Comment by donydchen** [comment]
Hi @crepuscularlight, thanks for your interest in our work.

If it works on pixelSplat, I assume your camera poses are aligned with the code base correctly, and the input views contain reasonable overlaps. Normally, it should also work correctly on MVSplat in this case.

Have you tried tuning the near and far in the depth range? You can check some more details at https://github.com/donydchen/mvsplat/issues/4#issuecomment-2019200882. Although it might contain artifacts, the correct near and far should give you reasonably nice-looking results when tested with the released models.

For some other related suggestions, you can check https://github.com/donydchen/mvsplat/issues/23.

Let me know if you try everything mentioned, but still nothing helps.

**Comment by crepuscularlight** [comment]
Thank you for your reply. After adjusting the near and far value, it works normally.

**Comment by CHDTevior** [comment]
> Thank you first for your great work. I try to adapt this model to dataset graspnet. And I follow the steps in pixelsplat to write the data loader. It works normally on pixelsplat but the training on mvsplat does not converge. Could you give me some advice how to debug this?

Hello, I also want to apply GraspNet dataset to pixelSplat and subsequent work. Can you share how you processed this dataset? Did you convert it into RealEstate10K format?

**Comment by CHDTevior** [comment]
> Thank you first for your great work. I try to adapt this model to dataset graspnet. And I follow the steps in pixelsplat to write the data loader. It works normally on pixelsplat but the training on mvsplat does not converge. Could you give me some advice how to debug this?

Hello, I have completed most of the work, but I found that the results I obtained are very strange, it seems that the camera extrinsics are not accurate. Can you tell me how you process the camera extrinsics of the GraspNet dataset?
![image](https://github.com/user-attachments/assets/e58192ec-0612-4bcc-bbde-2e57ba210934)
The above is the code for the camera parameters of the Kinect camera in each frame.

**Comment by crepuscularlight** [comment]
@CHDTevior Hello. I use realsense data and I convert it to re10k format. The code is `    intrinsic_path=example_path/"realsense"/"camK.npy"
    extrinsic_path = example_path/"realsense"/ "camera_poses.npy"
    transform_np = example_path / "realsense" / "cam0_wrt_table.npy"
    
    K_numpy = np.load(intrinsic_path)

    image=example_path / "realsense" /  "rgb" / "0000.png"
    img=Image.open(image)
    W,H=img.size
    K_numpy_normalized = K_numpy
    fx = K_numpy[0, 0] / W #fx
    cx = K_numpy[0, 2] / W #cx
    fy = K_numpy[1, 1] / H #fy
    cy = K_numpy[1, 2] / H #cy

    K_numpy_torch = torch.from_numpy(K_numpy_normalized).float()
    
    camera_pose_numpy = np.load(extrinsic_path)
    transform_np = np.load(transform_np)
    conversion = np.zeros((4, 4), dtype=np.float32)

    camera_pose_numpy = np.linalg.inv(transform_np @ camera_pose_numpy)
    # with example_path.open("r") as f:
    #     lines = f.read().splitlines()

    # url = lines[0]

    timestamps = range(camera_pose_numpy.shape[0])
    intrinsics = np.array([[fx,fy,cx,cy]],dtype=np.float32)
    intrinsics=np.repeat(intrinsics,camera_pose_numpy.shape[0],axis=0)

    camera_pose3x4=camera_pose_numpy[:,:3,:]
    cameras=np.zeros((camera_pose_numpy.shape[0],18))
    cameras[:,:4]=intrinsics
    cameras[:,6:]=camera_pose3x4.reshape((-1,12))
    # for line in lines[1:]:
    #     timestamp, *camera = line.split(" ")
    #     timestamps.append(int(timestamp))
    #     cameras.append(np.fromstring(",".join(camera), sep=","))

    # timestamps = torch.tensor(timestamps, dtype=torch.int64)
    cameras = torch.from_numpy(cameras)

    return {
        "timestamps":timestamps,
        "cameras": cameras,
    }`

**Comment by CHDTevior** [comment]
> @CHDTevior Hello. I use realsense data and I convert it to re10k format. The code is ` intrinsic_path=example_path/"realsense"/"camK.npy" extrinsic_path = example_path/"realsense"/ "camera_poses.npy" transform_np = example_path / "realsense" / "cam0_wrt_table.npy"
> 
> ```
> K_numpy = np.load(intrinsic_path)
> 
> image=example_path / "realsense" /  "rgb" / "0000.png"
> img=Image.open(image)
> W,H=img.size
> K_numpy_normalized = K_numpy
> fx = K_numpy[0, 0] / W #fx
> cx = K_numpy[0, 2] / W #cx
> fy = K_numpy[1, 1] / H #fy
> cy = K_numpy[1, 2] / H #cy
> 
> K_numpy_torch = torch.from_numpy(K_numpy_normalized).float()
> 
> camera_pose_numpy = np.load(extrinsic_path)
> transform_np = np.load(transform_np)
> conversion = np.zeros((4, 4), dtype=np.float32)
> 
> camera_pose_numpy = np.linalg.inv(transform_np @ camera_pose_numpy)
> # with example_path.open("r") as f:
> #     lines = f.read().splitlines()
> 
> # url = lines[0]
> 
> timestamps = range(camera_pose_numpy.shape[0])
> intrinsics = np.array([[fx,fy,cx,cy]],dtype=np.float32)
> intrinsics=np.repeat(intrinsics,camera_pose_numpy.shape[0],axis=0)
> 
> camera_pose3x4=camera_pose_numpy[:,:3,:]
> cameras=np.zeros((camera_pose_numpy.shape[0],18))
> cameras[:,:4]=intrinsics
> cameras[:,6:]=camera_pose3x4.reshape((-1,12))
> # for line in lines[1:]:
> #     timestamp, *camera = line.split(" ")
> #     timestamps.append(int(timestamp))
> #     cameras.append(np.fromstring(",".join(camera), sep=","))
> 
> # timestamps = torch.tensor(timestamps, dtype=torch.int64)
> cameras = torch.from_numpy(cameras)
> 
> return {
>     "timestamps":timestamps,
>     "cameras": cameras,
> }`
> ```

Thank you very much for your help. I have some questions about the hyperparameters during training. I set near and far to 400mm and 5900mm according to the gt value, which are 2.0 and 29.5 after scaling. I fine-tuned the model on re10k.ckpt. The rest of the training was the same as re10k. Could you tell me how many steps it took for the model to converge?
